Library/Homebrew/cask/audit.rb

```diff
@@ -410,17 +410,48 @@ def audit_token_valid
       add_error "cask token underscores should be replaced by hyphens" if cask.token.include? "_"
       add_error "cask token should not contain double hyphens" if cask.token.include? "--"
 
-      if cask.token.match?(/[^@a-z0-9-]/)
-        add_error "cask token should only contain lowercase alphanumeric characters, hyphens and @"
-      end
-
       if cask.token.start_with?("-", "@") || cask.token.end_with?("-", "@")
         add_error "cask token should not have leading or trailing hyphens and/or @"
       end
 
-      add_error "cask token @ unrelated to versioning should be replaced by -at-" if cask.token.count("@") > 1
-      add_error "cask token should not contain a hyphen followed by @" if cask.token.include? "-@"
-      add_error "cask token should not contain @ followed by a hyphen" if cask.token.include? "@-"
+      unversioned_token, _, version_designation = cask.token.rpartition("@")
+      if unversioned_token.empty?
+        match_data = /-(?<designation>alpha|beta|rc|release-candidate)$/.match(cask.token)
+        if match_data && cask.tap&.official? && cask.tap != "homebrew/cask-versions"
+          add_error "cask token should use @ before version designation '#{match_data[:designation]}'"
+        end
+        unversioned_token = cask.token
+        version_designation = ""
+      end
+
+      add_error "unversioned cask token @ should be replaced by -at-" if unversioned_token.include? "@"
```

Comment:
```suggestion
      add_error "cask token should use '-at-' instead of '@' in the unversioned part" if unversioned_token.include? "@"
```

Reply:
Thanks. I just realized that "unversioned cask" was a poor phrase given its existing usage for unversioned URLs (`bump-unversioned-casks`).